sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -685,8 +685,8 @@ public Read<T> withBeamSchemas(boolean withBeamSchemas) {
     @Override
     @SuppressWarnings("unchecked")
     public PCollection<T> expand(PBegin input) {
-      checkNotNull(getFilepattern(), "filepattern");
-      checkNotNull(getSchema(), "schema");
+      checkArgumentNotNull(getFilepattern(), "filepattern");
```

Comment:
These two should be `checkStateNotNull` because they are not referring to arguments.